Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -609,38 +609,62 @@ - (void)writeObjectToDefaults:(NSString *)key value:(NSObject *)value {
     [self persistPrefsToDisk];
 }
 
-- (void)persistPrefsToDisk {
-    NSDictionary *persistenceDict = [self.persistenceDict copy];
-    NSBlockOperation *newPersistOp = [NSBlockOperation blockOperationWithBlock:^{
-        if (![NSKeyedArchiver archiveRootObject:persistenceDict toFile:[self prefsFile]]) {
-            [self logWarning:@"Failed to persist preferences to disk"];
+- (void)persistPrefsToDisk
+    {
+    @synchronized (self)
+        {
+        NSDictionary *persistenceDict = [self.persistenceDict copy];
+        NSBlockOperation *newPersistOp = [NSBlockOperation blockOperationWithBlock:
+        ^   {
+            NSData *data = [NSKeyedArchiver archivedDataWithRootObject:persistenceDict];
+            if (!data)
+                {
```

Comment:
Hey Ed - This new `{` formatting syntax is quite a deviation from not only the rest of the SDK, but the rest of Branch's code styling around brackets. Might I ask that you keep with the same syntax for compatibility and readability?